backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java

```diff
@@ -50,65 +63,128 @@ private void updatePersonFacility(Person person, UUID facilityId) {
+            cb.or(
+                cb.isNull(root.get(SpecField.Facility)), // null check first
+                cb.equal(root.get(SpecField.Facility).get(SpecField.InternalId), facilityId)));
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public long getAllPatientsCount() {
-    return _repo.countAllByOrganization(_os.getCurrentOrganization(), false);
+  private Specification<Person> nameMatchesFilter(
+      @Size(min = MINIMUMCHARFORSEARCH) String namePrefixMatch) {
+    String likeString = namePrefixMatch.trim().toLowerCase() + "%";
+    return (root, query, cb) ->
+        cb.or(
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.FirstName)), likeString),
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.MiddleName)), likeString),
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.LastName)), likeString));
   }
 
-  // FYI archived is not just a parameter because it requires different permissions.
-  @AuthorizationConfiguration.RequirePermissionReadArchivedPatientList
-  public List<Person> getArchivedPatients(UUID facilityId, int pageOffset, int pageSize) {
-    return _repo
-        .findByFacilityAndOrganization(
-            _os.getFacilityInCurrentOrg(facilityId),
-            _os.getCurrentOrganization(),
-            true,
-            PageRequest.of(pageOffset, pageSize, NAME_SORT))
-        .toList();
+  private Specification<Person> isDeletedFilter(boolean isDeleted) {
+    return (root, query, cb) -> cb.equal(root.get(SpecField.IsDeleted), isDeleted);
+  }
+
+  // called by List function and Count function
+  private boolean checkPermissionsForListFunc(
+      UUID facilityId, boolean isArchived, String searchTerm) {
+    if (_auth.userHasSiteAdminRole()) { // site admins skip all other checks
+      return true;
+    }
+
+    ArrayList<UserPermission> perms = new ArrayList<UserPermission>();
+
+    if (facilityId == null) {
+      // READ_PATIENT_LIST does NOT seem right, probably an admin role to see all patients across
+      // facilities?
+      perms.add(UserPermission.READ_PATIENT_LIST);
+    }
+    if (isArchived) {
+      perms.add(UserPermission.READ_ARCHIVED_PATIENT_LIST);
+    }
+    if (searchTerm != null) {
+      perms.add(UserPermission.SEARCH_PATIENTS);
+    } else {
+      // because searchTerm is null, they are reading permissions across all patients
+      perms.add(UserPermission.READ_PATIENT_LIST);
+    }
+
+    // check all the permissions in one call.
+    return _auth.userHasPermissions(perms);
   }
 
-  @AuthorizationConfiguration.RequirePermissionReadArchivedPatientList
-  public long getArchivedPatientsCount(UUID facilityId) {
-    return _repo.countAllByFacilityAndOrganization(
-        _os.getFacilityInCurrentOrg(facilityId), _os.getCurrentOrganization(), true);
+  // called by List function and Count function
+  public Specification<Person> buildFilterForListFunc(
```

Comment:
Why public?

Reply:
Good catch. I think it's an artifact of transitory refactor (I originally tried to make this part of the Person object.)